lib/fs.js

```diff
@@ -412,7 +412,7 @@ function open(path, flags, mode, callback) {
   path = toPathIfFileURL(path);
   validatePath(path);
   const flagsNumber = stringToFlags(flags);
-  if (arguments.length < 4) {
+  if (typeof mode === 'function') {
```

Comment:
I think this will conflict with @bnoordhuis's work on https://github.com/nodejs/node/pull/23767

Reply:
Should we land #23767 first and then adapt this PR to that change? Seems do-able. /cc @codebytere @bnoordhuis